implementations/micrometer-registry-otlp/src/main/java/io/micrometer/registry/otlp/OtlpMeterRegistry.java

```diff
@@ -0,0 +1,261 @@
+        return new CumulativeFunctionCounter<>(id, obj, countFunction);
+    }
+
+    @Override
+    protected LongTaskTimer newLongTaskTimer(Meter.Id id, DistributionStatisticConfig distributionStatisticConfig) {
+        return new CumulativeHistogramLongTaskTimer(id, this.clock, getBaseTimeUnit(), distributionStatisticConfig);
+    }
+
+    @Override
+    protected TimeUnit getBaseTimeUnit() {
+        return TimeUnit.MILLISECONDS;
+    }
+
+    @Override
+    protected DistributionStatisticConfig defaultHistogramConfig() {
+        return DistributionStatisticConfig.builder()
+                .expiry(this.config.step())
+                .build()
+                .merge(DistributionStatisticConfig.DEFAULT);
+    }
+
+    Metric writeMeter(Meter meter) {
+        // TODO
+        return getMetricBuilder(meter.getId()).build();
+    }
+
+    Metric writeGauge(Gauge gauge) {
+        return getMetricBuilder(gauge.getId())
+                .setGauge(io.opentelemetry.proto.metrics.v1.Gauge.newBuilder()
+                        .addDataPoints(NumberDataPoint.newBuilder()
+                                .setTimeUnixNano(TimeUnit.MILLISECONDS.toNanos(this.clock.wallTime()))
+                                .setAsDouble(gauge.value())
+                                .addAllAttributes(getTagsForId(gauge.getId()))
+                                .build()))
+                .build();
+    }
+
+    Metric writeCounter(Counter counter) {
+        StartTimeAwareCumulativeCounter otlpCounter = (StartTimeAwareCumulativeCounter) counter;
+        return getMetricBuilder(otlpCounter.getId())
+                .setSum(Sum.newBuilder()
+                        .addDataPoints(NumberDataPoint.newBuilder()
+                                .setStartTimeUnixNano(otlpCounter.getStartTimeNanos())
+                                .setTimeUnixNano(TimeUnit.MILLISECONDS.toNanos(this.clock.wallTime()))
+                                .setAsDouble(counter.count())
+                                .addAllAttributes(getTagsForId(counter.getId()))
+                                .build())
+                        .setIsMonotonic(true)
+                        .setAggregationTemporality(AggregationTemporality.AGGREGATION_TEMPORALITY_CUMULATIVE)
+                        .build())
+                .build();
+    }
+
+    private Metric writeHistogramSupport(HistogramSupport histogramSupport) {
+        Metric.Builder metricBuilder = getMetricBuilder(histogramSupport.getId());
+        boolean isTimeBased = histogramSupport instanceof Timer;
+        HistogramSnapshot histogramSnapshot = histogramSupport.takeSnapshot();
+        long wallTimeNanos = TimeUnit.MILLISECONDS.toNanos(this.clock.wallTime());
+
+        HistogramDataPoint.Builder histogramDataPoint = HistogramDataPoint.newBuilder()
+                .addAllAttributes(getTagsForId(histogramSupport.getId()))
+                // TODO start time
+                .setTimeUnixNano(wallTimeNanos)
```

Comment:
What about the max and mins? I was curious to see if you were going to use the Measurement#getStatistic to populate this or do something different.

Reply:
max/min was added to the proto but hasn't been released yet, unfortunately: https://github.com/open-telemetry/opentelemetry-proto/compare/v0.16.0...main

Though as I was implementing this, there's a bit of a mismatch here and some choices. Min/max over the lifetime of the application don't seem particularly useful, but I think that's what we would be expected to send with cumulative temporality. I tried using delta temporality but the histogram didn't seem to get exported from the OTel collector to the Prometheus scape.

We've never supported a min, though it has been asked about for a backend that requires it for its histogram type. There's also the matter that we decay the max over multiple step intervals intentionally (reasoning described [here](https://github.com/micrometer-metrics/micrometer/issues/1993)), which is a different time window than the other stats. But we can only set one start/stop time for all of them.

Reply:
Sorry, I was just reading the proto directly and didn't consider that it hadn't been released yet. 

I don't have very strong opinions here, but I'm asking because we use a Micrometer meter registry today which publishes some of these using `statistic` as a tag. So I'll need to copy it forward or come up with a migration strategy.

Reply:
I'm not quite sure what we'll do, but we have some time to think about it, so I've marked it for follow-up after this pull request.